src/OpenTelemetry/Trace/SamplingParameters.cs

```diff
@@ -42,14 +43,14 @@ namespace OpenTelemetry.Trace
             ActivityTraceId traceId,
             string name,
             ActivityKind kind,
-            IEnumerable<KeyValuePair<string, object>> tags = null, // TODO: Empty
+            IEnumerable<KeyValuePair<string, object>> tags = null,
             IEnumerable<ActivityLink> links = null)
         {
             this.ParentContext = parentContext;
             this.TraceId = traceId;
             this.Name = name;
             this.Kind = kind;
-            this.Tags = tags;
+            this.Tags = tags ?? Enumerable.Empty<KeyValuePair<string, object>>();
```

Comment:
Is this considered a breaking change? If not provided, Tags used to be null. Now it'll be Empty collection. It makes writing sampler easy.
`foreach(var tag in tags) {}`
vs
`if (tags != null) {foreach(var tag in tags) {}}`

Reply:
Would this for-each introduce perf overhead?

Reply:
yes a minor one. (the null check would likely be the fastest, but empty is easy to write code)

I'll keep the existing `null` as it is possible that some is making sampling decision based on it being `null`, changing it can be breaking behavior.